src/Prism.Container.Extensions/IServiceProviderExtensions.cs

```diff
@@ -31,7 +31,7 @@ private static void RegisterTypesWithPrismContainer(IContainerRegistry container
                         else if (service.ImplementationInstance != null)
                             containerRegistry.RegisterInstance(service.ServiceType, service.ImplementationInstance);
                         else if (service.ImplementationFactory != null)
-                            containerRegistry.RegisterInstance(service.ServiceType, service.ImplementationFactory(containerRegistry as IServiceProvider));
+                            containerRegistry.RegisterDelegate(service.ServiceType, service.ImplementationFactory);
```

Comment:
It would actually be RegisterSingletonFromDelegate because it's under the Singleton case... that said already fixed...